osp/core/session/db/sql_wrapper_session.py

```diff
@@ -211,26 +295,26 @@ def _get_conditions(self, triple, table_name, object_datatype):
         s, p, o = triple
 
         if s is not None:
-            uid = to_uid(s)
+            uid = UID(s)
             conditions += [EqualsCondition("ts", "uid", uid,
-                                           "UID")]
+                                           UID.iri)]
 
         if p is not None and table_name != self.TYPES_TABLE:
             ns_idx, name = self._split_namespace(p)
             conditions += [
-                EqualsCondition("tp", "ns_idx", ns_idx, rdflib.XSD.integer),
-                EqualsCondition("tp", "name", name, rdflib.XSD.string)
+                EqualsCondition("tp", "ns_idx", ns_idx, XSD.integer),
+                EqualsCondition("tp", "name", name, XSD.string)
             ]
 
         if o is not None and table_name == self.RELATIONSHIP_TABLE:
```

Comment:
Do one `if o is not None` with nested ifs for the `table_name` to avoid checking `o` multiple times

Reply:
Done